signer/tests/integration/transaction_coordinator.rs

```diff
@@ -3075,3 +3083,420 @@ async fn test_get_btc_state_with_available_sweep_transactions_and_rbf() {
+        };
+        let counter = start_count.clone();
+        tokio::spawn(async move {
+            counter.fetch_add(1, Ordering::Relaxed);
+            ev.run().await
+        });
+
+        let ev = TxSignerEventLoop {
+            network: network.spawn(),
+            threshold: signatures_required as u32,
+            context: ctx.clone(),
+            context_window: 10000,
+            wsts_state_machines: LruCache::new(NonZeroUsize::new(100).unwrap()),
+            signer_private_key: kp.secret_key().into(),
+            rng: rand::rngs::OsRng,
+            dkg_begin_pause: None,
+        };
+        let counter = start_count.clone();
+        tokio::spawn(async move {
+            counter.fetch_add(1, Ordering::Relaxed);
+            ev.run().await
+        });
+
+        let ev = RequestDeciderEventLoop {
+            network: network.spawn(),
+            context: ctx.clone(),
+            context_window: 10000,
+            blocklist_checker: Some(()),
+            signer_private_key: kp.secret_key().into(),
+        };
+        let counter = start_count.clone();
+        tokio::spawn(async move {
+            counter.fetch_add(1, Ordering::Relaxed);
+            ev.run().await
+        });
+
+        let zmq_stream =
+            BitcoinCoreMessageStream::new_from_endpoint(BITCOIN_CORE_ZMQ_ENDPOINT, &["hashblock"])
+                .await
+                .unwrap();
+        let (sender, receiver) = tokio::sync::mpsc::channel(100);
+
+        tokio::spawn(async move {
+            let mut stream = zmq_stream.to_block_hash_stream();
+            while let Some(block) = stream.next().await {
+                sender.send(block).await.unwrap();
+            }
+        });
+
+        let block_observer = BlockObserver {
+            context: ctx.clone(),
+            bitcoin_blocks: ReceiverStream::new(receiver),
+        };
+        let counter = start_count.clone();
+        tokio::spawn(async move {
+            counter.fetch_add(1, Ordering::Relaxed);
+            block_observer.run().await
+        });
+    }
+
+    while start_count.load(Ordering::SeqCst) < 12 {
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+
+    // =========================================================================
+    // Wait for the first signer to be the coordinator
+    // -------------------------------------------------------------------------
+    // - Two of three signers will not be able to coordinate, because of failing
+    //   in getting deposits (so no signal is sent to request decider and tx
+    //   coordinator)
+    // =========================================================================
+    let signers_key = setup.signers.signer_keys().iter().cloned().collect();
+    loop {
+        let chain_tip: BitcoinBlockHash = faucet.generate_blocks(1).pop().unwrap().into();
+
+        if given_key_is_coordinator(signers[0].2.public_key().into(), &chain_tip, &signers_key) {
+            break;
+        }
+    }
```

Comment:
We could also just loop through all signers and break when we figure out who the coordinator is and not their index. I just want to limit the number of bitcoin blocks are created, since that slows down tests. Well, at least until Cyle's containerized CI PR lands.

Reply:
Oh wait, this isn't so straightforward. Okay let me think.

Reply:
Yeah, this logic seems required given what we want to test. Let's roll with what is here and try to change to each test having their own bitcoin node container. If we cannot do that for some reason then we can figure something else out.

Reply:
Yeah this is definitely suboptimal, we could try to figure it out beforehand who is the coordinator or change the guard on the limits mock to be able to set it dynamically, but given the CI/integration tests improvements on the horizon I would +1 on waiting to see if things are bad and we need to address this. Also this test uses 3 signers so it shouldn't require too many blocks before getting the right coordinator.